src/resolve/resolved-bus.c

```diff
@@ -1581,6 +1583,174 @@ static int bus_method_reset_server_features(sd_bus_message *message, void *userd
         return sd_bus_reply_method_return(message, NULL);
 }
 
+static int on_bus_track(sd_bus_track *t, void *userdata) {
+        DnssdService *s = userdata;
+
+        assert(t);
+        assert(s);
+
+        log_debug("Client of active request vanished, destroying DNS-SD service.");
+        dnssd_service_free(s);
+
+        return 0;
+}
+
+static int bus_method_register_service(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_(dnssd_service_freep) DnssdService *service = NULL;
+        _cleanup_(sd_bus_track_unrefp) sd_bus_track *bus_track = NULL;
+        _cleanup_free_ char *path = NULL;
+        _cleanup_free_ char *instance_name = NULL;
+        Manager *m = userdata;
+        DnssdService *s = NULL;
+        DnsTxtItem *last = NULL;
+        const char *name;
+        const char *name_template;
+        const char *type;
+        int r;
+
+        assert(message);
+        assert(m);
+
+        if (m->mdns_support != RESOLVE_SUPPORT_YES)
+                return sd_bus_error_setf(error, SD_BUS_ERROR_NOT_SUPPORTED, "Support for MulticastDNS is disabled");
+
+        service = new0(DnssdService, 1);
+        if (!service)
+                return sd_bus_error_setf(error, SD_BUS_ERROR_NO_MEMORY, "Out of memory");
```

Comment:
oom errors are really server-side problems, hence I figure we should just do "return log_oom()" here, so that we actually log this locally instead pretending this was a client-side problem and just returning it to the user, as if the user could do anything about it...

by doing "return log_oom()" we actually do both: return an error to the client, and log about it as server problem, and that's good.